### PR TITLE
fix-get-parent-properties

### PR DESCRIPTION
**Pull request by qiaozha**
fixes https://github.com/Azure/autorest.typescript/issues/2014

### CODE REVIEW

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts [hunk 1]
```diff
@@ -943,3 +943,15 @@ export function hasPagingOperation(codeModel: ModularCodeModel) {
     )
   );
 }
+
+function getAllProperties(type: Type): Property[] {
+  const properties: Property[] = [];
+  if (!type) {
+    return properties;
+  }
+  type.parents?.forEach((p) => {
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/helpers/operationHelpers.ts:952]: We should also consider the multiple hierarchy.

A -> B -> C.

File: packages/typespec-ts/test/modularUnit/modelsGenerator.spec.ts [hunk 2]
```diff
@@ -804,4 +804,83 @@ describe("inheritance & polymorphism", () => {
       }`
     );
   });
+
+  it.only("should handle inheritance model in operations", async () => {
```
Comment (MaryGao) [inline packages/typespec-ts/test/modularUnit/modelsGenerator.spec.ts:808]: skip `only` & could you add test cases for A -> B -> C with different discriminator?

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts [hunk 3]
```diff
@@ -943,3 +943,15 @@ export function hasPagingOperation(codeModel: ModularCodeModel) {
     )
   );
 }
+
+function getAllProperties(type: Type): Property[] {
+  const properties: Property[] = [];
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/helpers/operationHelpers.ts:948]: nit: not sure this is caused the duplicated properties issues when discriminator property exists both parent and child model. see below cadl-ranch cases:

https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/type/model/inheritance/single-discriminator/main.tsp#L12-L25C1

**Review by MaryGao — APPROVED** [review]
Left small comments.

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts (hunk 3 above)
Comment (qiaozha) [reply packages/typespec-ts/src/modular/helpers/operationHelpers.ts:948]: Yeah, I found that, when I was trying to enable cadl ranch test cases for model inheritance in modular.

**Pull request merged**